pkg/graveler/graveler.go

```diff
@@ -1526,6 +1553,11 @@ func (g *KVGraveler) deleteUnsafe(ctx context.Context, repository *RepositoryRec
 	return g.StagingManager.Set(ctx, branch.StagingToken, key, nil, true)
 }
 
+// ListStaging Exposing listStagingArea to catalog for PrepareGCUncommitted
+func (g *KVGraveler) ListStaging(ctx context.Context, branch *Branch) (ValueIterator, error) {
```

Comment:
You can just expose the original method.

Reply:
I prefer not exposing the original method. It was created for a specific reason which is unrelated to GC. Any additional logic that GC may require should be added to the new method.